yasnippet.el

```diff
@@ -4240,7 +4240,13 @@ When multiple expressions are found, only the last one counts."
                                                                         (yas--mirror-end mirror)))))
       (goto-char (yas--mirror-start mirror))
       (let ((yas--inhibit-overlay-hooks t))
-        (insert reflection))
+        (insert reflection)
+        (let ((start (yas--mirror-start mirror))
+              (end (yas--mirror-end mirror)))
+          (when (and (eq yas-indent-line 'auto)
+                     (not (eq (line-number-at-pos start)
+                              (line-number-at-pos end))))
+            (indent-region start end))))
```

Comment:
We either don't care about indenting the first line in which case we can just indent all of the lines unconditionally. Or else we do care, and we should make sure only to indent the last n-1 lines.